fix(seat-reservation-web-component): make build in windows running

sk-sbb opened the pull request:
## Preflight Checklist

<!-- Please ensure you've completed the following steps by replacing [ ] with [x]-->

- [ ] I have read the [Contributing Guidelines](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CONTRIBUTING.md) for this project.
- [ ] I agree to follow the [Code of Conduct](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CODE_OF_CONDUCT.md) that this project adheres to.
- [ ] I have searched the [pull request tracker](https://github.com/sbb-design-systems/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../docs/REVIEW.md) for more information on what is checked during the review process.

## Changes

Changes in this pull request:

-

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [ ] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

<!-- Please do not submit updates to dependencies unless it fixes an issue. -->

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. -->

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

<!-- Any other information that is important to this PR such as screenshots of how the component looks before and after the change. -->

kyubisation reviewed (changes requested):
Hello there 👋 
Could you explain why this is necessary?
The configuration for externals should not be required, as this is already covered in the tools/web-test-runner/vite-plugin.ts for testing and rollupOptions is only applied to library code and therefore should not be necessary.
Additionally, it is not documented and unclear to me why the top level await module is added.

sk-sbb commented:
hi @kyubisation,
it was more or less trial and error approach to get yarn build running on windows.
Initially, build has failed with following error:
`error during build:
[vite]: Rollup failed to resolve import "/__web-dev-server__web-socket.js" from "C:/devsbb/pr/gh/lyne-components/node_modules/@web/test-runner-commands/browser/commands.mjs".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
build.rollupOptions.external`

I looked up how "__web-dev-server__web-socket.js" is treated across the project and stumbled accross vite.plugin.ts in web-test-runner, where it has been treated as external. I assumed that this exteranlization applies only in tests, so I added similar snippet to vite.config.js in elements folder.

After that the fehler was gone, but the next one appeared:
`error during build:
[vite:esbuild-transpile] Transform failed with 1 error:
core\testing\private\test-setup.js:61:0: ERROR: Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
file: core\testing\private\test-setup.js:61:0`
which I was able to get rid of after asking google and using this plugin :)

Althou it does look like curing symptoms instead of an illness itself, I introduced the solution to my team - and they meant we could make a PR out of this, so that you can evaluate if this solution is worth of an attempt.

And yes I should have included some explanation to PR itself :)
Thanks for taking your time!

kyubisation commented:
I'll try to have a look at this as soon as I have some time to spare. I apologize, as this might take a while due to the current major release preparation.
Can you try to fix the build in the meantime?
